src/ccutil/ccutil.cpp

```diff
@@ -33,68 +32,72 @@ CCUtil::CCUtil()
   if (!argv0.empty()) {
-    /* Use tessdata prefix from the command line. */
-    datadir = argv0;
-  } else if (tessdata_prefix) {
-    /* Use tessdata prefix from the environment. */
-    datadir = tessdata_prefix;
-#if defined(_WIN32)
-  } else if (datadir.empty() || !std::filesystem::exists(datadir)) {
-    /* Look for tessdata in directory of executable. */
-    char path[_MAX_PATH];
-    DWORD length = GetModuleFileName(nullptr, path, sizeof(path));
-    if (length > 0 && length < sizeof(path)) {
-      char *separator = std::strrchr(path, '\\');
-      if (separator != nullptr) {
-        *separator = '\0';
-        std::string subdir = path;
-        subdir += "/tessdata";
-        if (std::filesystem::exists(subdir)) {
-          datadir = subdir;
-        }
-      }
+    std::filesystem::path path(argv0);
+    if (!std::filesystem::is_directory(path)) {
+      tesserr << "Warning (tessdata): '" << argv0 << "' is not a valid directory.\n";
     }
-#endif /* _WIN32 */
+    return path;
   }
 
-  // datadir may still be empty:
-  if (datadir.empty()) {
-#if defined(TESSDATA_PREFIX)
-    // Use tessdata prefix which was compiled in.
-    datadir = TESSDATA_PREFIX "/tessdata/";
-    // Note that some software (for example conda) patches TESSDATA_PREFIX
-    // in the binary, so it might be shorter. Recalculate its length.
-    datadir.resize(std::strlen(datadir.c_str()));
-#else
-    datadir = "./";
-#endif /* TESSDATA_PREFIX */
+  // Check environment variable if argv0 is not specified
+  if (const char *tessdata_prefix = std::getenv("TESSDATA_PREFIX")) {
+    std::filesystem::path path(tessdata_prefix);
+    if (std::filesystem::exists(path)) {
+      return path;
+    } else {
+      tprintf("Warning: TESSDATA_PREFIX %s does not exist, ignoring.\n",
+              tessdata_prefix);
+    }
   }
 
-  // check for missing directory separator
-  const char lastchar = datadir.back();
-  if (lastchar != '/' && lastchar != '\\') {
-    datadir += '/';
+#ifdef _WIN32
+  // Windows-specific: check for 'tessdata' not existing in the executable
+  // directory
+  wchar_t path[MAX_PATH];
+  if (DWORD length = GetModuleFileNameW(nullptr, path, MAX_PATH);
+      length > 0 && length < MAX_PATH) {
+    std::filesystem::path exe_path(path);
+    auto tessdata_subdir = exe_path.parent_path() / "tessdata";
+    if (std::filesystem::exists(tessdata_subdir)) {
+      return tessdata_subdir;
+    }
   }
+#endif
+
+  // Fallback to compile-time or current directory
+#ifdef TESSDATA_PREFIX
+  return std::filesystem::path(TESSDATA_PREFIX) / "tessdata";
+#else
+  return std::filesystem::current_path();
+#endif
 }
 
+
+/**
+ * @brief CCUtil::main_setup - set location of tessdata and name of image
+ *
+ * @param argv0 - paths to the directory with language files and config files.
+ */
+void CCUtil::main_setup(const std::string &argv0, const std::string &basename) {
+  imagebasename = basename; /**< name of image */
+  datadir = find_data_path(argv0);
```

Comment:
The old `datadir` was always terminated with a '/' which is missing for the new datadir`. This is an API change, and it might also be the reason for the failing CI tests.